component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateServlet.java

```diff
@@ -69,6 +71,21 @@ protected void doPost(HttpServletRequest servletRequest, HttpServletResponse ser
             throw new RuntimeException("Error occurred while building URL.", e);
         }
 
+        String authenticationEndpoint = X509CertificateUtil.getX509Parameters().get(X509CertificateConstants
+                .AUTHENTICATION_ENDPOINT_PARAMETER);
+
+        if (authenticationEndpoint != null) {
+            URL authenticationEndpointURL = new URL(authenticationEndpoint);
+            String protocol = authenticationEndpointURL.getProtocol();
+            String host = authenticationEndpointURL.getHost();
+            int port = authenticationEndpointURL.getPort();
+            if (port == -1) {
+                commonAuthURL = protocol + "://" + host + "/" + X509CertificateConstants.COMMON_AUTH;
+            } else {
+                commonAuthURL = protocol + "://" + host + ":" + port + "/" + X509CertificateConstants.COMMON_AUTH;
```

Comment:
AFAIU the commonAuthURL will not contain the port number if the port is only `-1`. Can this port be a different minus value like `-2`, `-3` etc. If so is it ok to add that to the URL?

Reply:
According to the documentation of getPort[1] method, `Returns: the port number, or -1 if the port is not set`.

[1] https://docs.oracle.com/javase/8/docs/api/java/net/URL.html#getPort--